### PR TITLE
Validation of behavior and implementation modules

### PR DESCRIPTION
**Pull request by saneery**
:wave:
relates to https://github.com/msz/hammox/issues/22

### CODE REVIEW

**Comment by saneery** [comment]
@msz Hi! How can I trigger CI?

File: lib/hammox.ex [hunk 1]
```diff
@@ -333,16 +342,20 @@ defmodule Hammox do
...
+        callbacks
+        |> Enum.find_value([], fn
+          {{^function_name, ^arity}, typespecs} -> typespecs
+          _ -> false
+        end)
+        |> Enum.map(fn typespec ->
+          Utils.replace_user_types(typespec, behaviour_module_name)
+        end)
+
+      :error ->
```
Comment (msz) [inline lib/hammox.ex:356]: What is the reason behind handling `:error` here? I don't see a use case for this function to silently proceed when the `Code.Typespec.fetch_callbacks(behaviour_module_name)` call errors — I'd rather make it fail to be stricter.

**Review by msz — COMMENTED** [review]
Great work, thank you! Just a small comment.

**Comment by msz** [comment]
Don't worry about CI, it doesn't run on PRs from forks for security

File: lib/hammox.ex (hunk 1 above)
Comment (saneery) [reply lib/hammox.ex:356]: Actually, looks like the `Code.Typespec.fetch_callbacks(behaviour_module_name)` calls error only then the module doesn't exist. So I think there is a need to add validation of behavior module inside `protect/2` function and raise an exception.

**Comment by msz** [comment]
I merged #51, could you rebase and resolve the conflict?

**Review by msz — APPROVED** [review]

**Comment by saneery** [comment]
Done! :)

**Comment by saneery** [comment]
@msz could you add `hacktoberfest-accepted` labels to my PRs? :)

**Pull request merged**